### PR TITLE
Add library to encode/decode Gorilla compressed buffers.

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

We use a custom format but the idea remains the same. In our case,
we don't have timestamps so we only XOR values.

##### Test Plan

- Run it with libFuzzy for quite a while.
- CI jobs.

### CODE REVIEW

**Comment by vkalintiris** [comment]
Google's benchmark library reports the following when encoding/decoding 1024 elements in a single core:

```
Run on (16 X 2000 MHz CPU s)
CPU Caches:
  L1 Data 64 KiB (x16)
  L1 Instruction 64 KiB (x16)
  L2 Unified 512 KiB (x16)
  L3 Unified 16384 KiB (x1)
Load Average: 0.11, 0.33, 0.19
------------------------------------------------------------------------------
Benchmark                    Time             CPU   Iterations UserCounters...
------------------------------------------------------------------------------
BM_EncodeU32Numbers       9099 ns         9099 ns        76953 bytes_per_second=429.288M/s items_per_second=112.535M/s
BM_DecodeU32Numbers       8714 ns         8714 ns        79933 bytes_per_second=448.253M/s items_per_second=117.507M/s
BM_EncodeU64Numbers       7867 ns         7867 ns        88181 bytes_per_second=993.072M/s items_per_second=130.164M/s
BM_DecodeU64Numbers       8180 ns         8180 ns        85790 bytes_per_second=955.067M/s items_per_second=125.183M/s
```

File: libnetdata/gorilla/gorilla.cc [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+    bit_code_t<Word> bcr;
+
+    bit_code_init(&bcr, (Word *) src, src_len);
+
+    Word num_entries;
+    if (!bit_code_info(&bcr, &num_entries, (Word *) NULL)) {
+        return 0;
+    }
+    if (num_entries > dst_len) {
+        return false;
```
Comment (thiagoftsm) [inline libnetdata/gorilla/gorilla.cc:322]: `false` is zero, but considering that this is the unique return that is not explicitly returning a size, I suggest to return 0 here too.

File: libnetdata/gorilla/gorilla.cc [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+bool bit_code_reader_u64_read(bit_code_reader_u64_t *bcr, uint64_t *number) {
+    bit_code_t<uint64_t> *bc = (bit_code_t<uint64_t> *) bcr;
+    return bit_code_read(bc, number);
+}
+
+bool bit_code_reader_u64_info(bit_code_reader_u64_t *bcr, uint64_t *num_entries_written,
+                                                          uint64_t *num_bits_written) {
+    bit_code_t<uint64_t> *bc = (bit_code_t<uint64_t> *) bcr;
+    return bit_code_info(bc, num_entries_written, num_bits_written);
+}
```
Comment (thiagoftsm) [inline libnetdata/gorilla/gorilla.cc:401]: @vkalintiris both 32bit and 64bit functions have more or less same algorithm, but we are not using `function overload`, please, can you confirm we will expose this code for other languages? If it is not the case, it is not clear the motive we are not using `function overload`.

Comment (vkalintiris) [reply libnetdata/gorilla/gorilla.cc:403]: I want to give @ktsaou the flexibility of picking whichever version he want from code that will be written in C.

**Review by stelfrag — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]

**Pull request merged**